src/components/CloudinaryContext/CloudinaryContext.js

```diff
@@ -24,35 +15,32 @@ const getNonCloudinaryProps = (props) => {
  *
  */
 class CloudinaryContext extends CloudinaryComponent {
-  constructor(props) {
-    super(props);
+  constructor(props, context) {
+    super(props, context);
+
     this.calcState = this.calcState.bind(this);
     this.state = this.calcState();
   }
 
   calcState() {
-    this.context = this.context || {};
-    let context = {};
-    const camelProps = Util.withCamelCaseKeys(this.props);
-
-    // only pass valid Cloudinary options
-    CloudinaryComponent.VALID_OPTIONS.forEach(key => {
-      let val = camelProps[key] || this.context[key];
-      if (val !== undefined && val !== null) {
-        context[key] = val;
-      }
-    });
-    return context;
+    const mergedProps = {...(this.context || {}), ...this.props};
+    const context = cloudinaryProps(mergedProps, CloudinaryComponent.VALID_OPTIONS);
+    const {includeOwnBody, ...childrenProps} = nonCloudinaryProps(this.props, CloudinaryContext.CLOUDINARY_PROPS);
```

Comment:
Again, will have a single util to separate the cloudinary from the non-cloudinary props